aten/src/ATen/native/layer_norm.cpp

```diff
@@ -0,0 +1,148 @@
+#include <ATen/NativeFunctions.h>
+
+#include <functional>
+#include <numeric>
+#include <tuple>
+#include <vector>
+
+#include <ATen/ATen.h>
+#include <ATen/AccumulateType.h>
+#include <ATen/CPUApplyUtils.h>
+#include <ATen/Config.h>
+#include <ATen/Parallel.h>
+
+namespace at {
+namespace native {
+
+namespace {
+
+template <typename T>
+void ElementwiseAffine(
+    const std::int64_t N,
+    const T mean,
+    const T rstd,
+    const T* X,
+    const T* gamma,
+    const T* beta,
+    T* Y) {
+  const T scale = rstd;
+  const T bias = -rstd * mean;
+  if (gamma != nullptr && beta != nullptr) {
+    for (std::int64_t i = 0; i < N; ++i) {
+      Y[i] = (X[i] * scale + bias) * gamma[i] + beta[i];
+    }
+  } else if (gamma != nullptr) {
+    for (std::int64_t i = 0; i < N; ++i) {
+      Y[i] = (X[i] * scale + bias) * gamma[i];
+    }
+  } else if (beta != nullptr) {
+    for (std::int64_t i = 0; i < N; ++i) {
+      Y[i] = X[i] * scale + bias + beta[i];
+    }
+  } else {
+    for (std::int64_t i = 0; i < N; ++i) {
+      Y[i] = X[i] * scale + bias;
+    }
+  }
+}
+
+template <typename T>
+std::tuple<Tensor, Tensor, Tensor> LayerNormForwardCPUImpl(
+    const Tensor& X,
+    const Tensor& gamma,
+    const Tensor& beta,
+    const int axis,
+    const T eps) {
+  const std::vector<std::int64_t> X_dims = X.sizes().vec();
+  const std::vector<std::int64_t> outer_dims(
+      X_dims.cbegin(), X_dims.cbegin() + axis);
+  const std::int64_t M = std::accumulate(
+      X_dims.cbegin(),
+      X_dims.cbegin() + axis,
+      1LL,
+      std::multiplies<std::int64_t>());
+  const std::int64_t N = std::accumulate(
+      X_dims.cbegin() + axis,
+      X_dims.cend(),
+      1LL,
+      std::multiplies<std::int64_t>());
+  Tensor Y = at::empty_like(X);
```

Comment:
I'm quite sure you want to call at:native::empty_like here and avoid all tracing/profiling checks.